src/util.rs

```diff
@@ -802,27 +802,31 @@ impl Run {
                 )
             }
             RunMode::Egglog => {
-                assert_eq!(self.eggcc_config.schedule, Schedule::Parallel, "Output egglog only works in parallel mode. Sequential mode does not use a single egraph");
-
                 let rvsdg = Optimizer::program_to_rvsdg(&self.prog_with_args.program)?;
                 let (dag, mut cache) = rvsdg.to_dag_encoding(true);
 
-                let schedule_steps = parallel_schedule();
-                if schedule_steps.len() != 1 {
-                    log::warn!("Parallel schedule had multiple steps! You may need to adjust the schedule to make eggcc tractable.");
-                }
+                let eggcc_config = EggccConfig {
+                    // stop before the last pass.
+                    stop_after_n_passes: -2,
+                    .. self.eggcc_config.clone()
+                };
+                let optimized = dag_in_context::optimize(&dag, &mut cache, &eggcc_config)
```

Comment:
You could refactor the inner loop of optimize into a function so you can get the egglog file directly instead of calling build_program

Reply:
hmm, I'm not sure how to refactor here. Maybe leave it to a future PR?

Reply:
sure